File: WebApplication/src/modules/user/models/user.dto.ts
import { AuthorityDto } from "../../nomenclature/dtos/authority.dto";
import { UserStatus } from "../enums/user-status.enum";
import { RoleDto } from "./role.dto";

export class UserDto {
  id: number;
  username: string;
  password: string;
  firstName: string;
  middleName: string;
  lastName: string;
  email: string;
  phone: string;
  role: RoleDto;
  status: UserStatus;
  authorityId: number;
  authority: AuthorityDto;
  position: string;
}

File: Zopoesht.Application/Users/Dtos/UserDto.cs
using Zopoesht.Application.Nomenclatures.Dtos;
using Zopoesht.Data.Users.Enums;

namespace Zopoesht.Application.Users.Dtos
{
    public class UserDto
    {
        public int Id { get; set; }

        public string FirstName { get; set; }

        public string MiddleName { get; set; }

        public string LastName { get; set; }

        public string Username { get; set; }

        public string Password { get; set; }

        public string Phone { get; set; }

        public string Email { get; set; }

        public RoleDto Role { get; set; }

        public int?  AuthorityId { get; set; }
        public AuthorityDto Authority { get; set; }

        public UserStatus Status { get; set; }

        public string Position { get; set; }
    }
}
